doc/modules/classes.rst

```diff
@@ -1349,3 +1355,67 @@ Low-level methods
    utils.estimator_checks.check_estimator
    utils.resample
    utils.shuffle
+
+
+Recently deprecated
```

Comment:
@waterponey @amueller Do you think it would be nice to link this to a more accessible section under whatsnew?